pkg/server/plugin/bundlepublisher/awsrolesanywhere/awsrolesanywhere_test.go

```diff
@@ -0,0 +1,376 @@
+			t:                   t,
+			expectTrustAnchorID: aws.String(config.TrustAnchorID),
+		}, nil
+	}
+	p := newPlugin(newClient)
+	plugintest.Load(t, builtin(p), nil, options...)
+	require.NoError(t, err)
+
+	// Test multiple update trust anchor operations, and check that only a call to
+	// UpdateTrustAnchor is made when there is a modified bundle that was not successfully
+	// published before.
+
+	// Have an initial bundle with SequenceNumber = 1.
+	bundle := getTestBundle(t)
+	bundle.SequenceNumber = 1
+
+	client, ok := p.rolesAnywhereClient.(*fakeClient)
+	require.True(t, ok)
+
+	// Reset the API call counters.
+	client.updateTrustAnchorCount = 0
+
+	// Throw an error when calling UpdateTrustAnchor.
+	client.updateTrustAnchorErr = errors.New("error calling UpdateTrustAnchor")
+
+	// Call PublishBundle. UpdateTrustAnchor should be called and return an error.
+	resp, err := p.PublishBundle(context.Background(), &bundlepublisherv1.PublishBundleRequest{
+		Bundle: bundle,
+	})
+	require.Error(t, err)
+	require.Nil(t, resp)
+
+	// UpdateTrustAnchor was called, even though it failed, so its counter should be incremented.
+	require.Equal(t, 1, client.updateTrustAnchorCount)
+
+	// Remove the updateTrustAnchorErr and try again.
+	client.updateTrustAnchorErr = nil
+	resp, err = p.PublishBundle(context.Background(), &bundlepublisherv1.PublishBundleRequest{
+		Bundle: bundle,
+	})
+	require.NoError(t, err)
+	require.NotNil(t, resp)
+	require.Equal(t, 2, client.updateTrustAnchorCount)
+
+	// Call PublishBundle with the same bundle.
+	resp, err = p.PublishBundle(context.Background(), &bundlepublisherv1.PublishBundleRequest{
+		Bundle: bundle,
+	})
+	require.NoError(t, err)
+	require.NotNil(t, resp)
+
+	// The same bundle was used, the counter should be the same as before.
+	require.Equal(t, 2, client.updateTrustAnchorCount)
+
+	// Have a new bundle and call PublishBundle.
+	bundle = getTestBundle(t)
+	bundle.SequenceNumber = 2
+	resp, err = p.PublishBundle(context.Background(), &bundlepublisherv1.PublishBundleRequest{
+		Bundle: bundle,
+	})
+	require.NoError(t, err)
+	require.NotNil(t, resp)
+
+	// PublishBundle was called with a different bundle, updateTrustAnchorCount should
+	// be incremented to be 3.
+	require.Equal(t, 3, client.updateTrustAnchorCount)
+
+	// Try to publish a bundle that's too large, and expect that we receive an error.
+	bundle = getLargeTestBundle(t)
+	bundle.SequenceNumber = 3
+	resp, err = p.PublishBundle(context.Background(), &bundlepublisherv1.PublishBundleRequest{
+		Bundle: bundle,
+	})
+	require.Error(t, err)
+}
+
+type fakeClient struct {
+	t *testing.T
+
+	awsConfig              aws.Config
+	updateTrustAnchorErr   error
+	updateTrustAnchorCount int
+
+	expectTrustAnchorID *string
+}
+
+func (c *fakeClient) UpdateTrustAnchor(_ context.Context, params *rolesanywhere.UpdateTrustAnchorInput, _ ...func(*rolesanywhere.Options)) (*rolesanywhere.UpdateTrustAnchorOutput, error) {
+	c.updateTrustAnchorCount++
```

Comment:
The updateTrustAnchorCount counter should be incremented only if there is no error.
```suggestion
```